### ISSUE TITLE
airnode-validator fails to build with the newly published @api3/ois package

### ISSUE THREAD

**Issue by aquarat** [api3dao/airnode#1422]
## Description

airnode-validator fails to build with the newly published @api3/ois package.

## Version

Please provide your Node.js, Yarn and relevant Airnode versions. Are you building from a branch, which commit?
Node 14.19.3
Yarn 1.22.19
master branch with ois dependency bumped (5acdba4f88ffeb219fdd47d615c2c85ab725825e with patch included below applied)

## Present Behaviour

airnode-validator fails to build with the newly published @api3/ois package.

## Expected Behaviour

The build should succeed.

## Steps to reproduce

Bump @api3/ois dependency to the latest in packages in Airnode that use it and then attempt the build.

## Additional comments:
It probably relates to @api3dao/ois issue 

The build log:
```
In airnode, commit 5acdba4f88ffeb219fdd47d615c2c85ab725825e:
yarn bootstrap
yarn build
...
yarn run v1.22.19
$ lerna run build --stream && ts-node scripts/make-cli-executable.ts
lerna notice cli v5.5.0
lerna info Executing command in 10 packages: "yarn run build"
@api3/airnode-abi: $ yarn run clean && yarn run compile
@api3/airnode-adapter: $ yarn run clean && yarn run compile
@api3/airnode-validator: $ yarn run clean && yarn run compile
@api3/airnode-protocol: $ yarn run clean && yarn compile
@api3/airnode-abi: $ rimraf -rf *.tsbuildinfo ./dist ./build *.tgz
@api3/airnode-protocol: $ rimraf -rf ./src/contracts *.tsbuildinfo ./dist ./build *.tgz
@api3/airnode-validator: $ rimraf -rf *.tsbuildinfo ./dist ./build *.tgz
@api3/airnode-adapter: $ rimraf -rf *.tsbuildinfo ./dist ./build *.tgz
@api3/airnode-abi: $ tsc --build tsconfig.json
@api3/airnode-protocol: $ yarn run compile:contracts && yarn run compile:tsc
@api3/airnode-validator: $ yarn tsc --build tsconfig.json
@api3/airnode-adapter: $ hardhat compile && tsc --build tsconfig.json
@api3/airnode-protocol: $ hardhat compile
@api3/airnode-validator: $ /Users/aquarat/Projects/airnode/node_modules/.bin/tsc --build tsconfig.json
@api3/airnode-protocol: Nothing to compile
@api3/airnode-protocol: $ yarn compile:contract-dts && yarn compile:copy-contract-dts && tsc --build tsconfig.json
@api3/airnode-protocol: $ yarn typechain --target ethers-v5 --out-dir ./src/contracts "./artifacts/contracts/**/!(*.dbg).json"
@api3/airnode-protocol: $ /Users/aquarat/Projects/airnode/node_modules/.bin/typechain --target ethers-v5 --out-dir ./src/contracts './artifacts/contracts/**/!(*.dbg).json'
@api3/airnode-adapter: Nothing to compile
@api3/airnode-protocol: Successfully generated 113 typings!
@api3/airnode-protocol: $ copyfiles -u 1 "src/contracts/**/*.d.ts" dist/src
@api3/airnode-validator: src/config/config.ts(425,10): error TS2589: Type instantiation is excessively deep and possibly infinite.
@api3/airnode-validator: src/config/config.ts(425,10): error TS2589: Type instantiation is excessively deep and possibly infinite.
@api3/airnode-validator: src/config/config.ts(425,10): error TS2589: Type instantiation is excessively deep and possibly infinite.
@api3/airnode-validator: error Command failed with exit code 1.
@api3/airnode-validator: info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
@api3/airnode-validator: error Command failed with exit code 1.
@api3/airnode-validator: info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
@api3/airnode-validator: error Command failed with exit code 1.
@api3/airnode-validator: info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
lerna ERR! yarn run build exited 1 in '@api3/airnode-validator'
error Command failed with exit code 1.
```

Patch:
```
diff --git a/packages/airnode-adapter/package.json b/packages/airnode-adapter/package.json
index 34ec0a22..7f6005fe 100644
--- a/packages/airnode-adapter/package.json
+++ b/packages/airnode-adapter/package.json
@@ -19,7 +19,7 @@
     "test:watch": "yarn test:ts --watch"
   },
   "dependencies": {
-    "@api3/ois": "^1.1.1",
+    "@api3/ois": "^1.1.2",
     "@api3/promise-utils": "^0.3.0",
     "axios": "^0.27.2",
     "bignumber.js": "^9.1.0",
diff --git a/packages/airnode-node/package.json b/packages/airnode-node/package.json
index 68439ecb..350712b7 100644
--- a/packages/airnode-node/package.json
+++ b/packages/airnode-node/package.json
@@ -29,7 +29,7 @@
     "@api3/airnode-protocol": "^0.7.1",
     "@api3/airnode-utilities": "^0.7.1",
     "@api3/airnode-validator": "^0.7.1",
-    "@api3/ois": "^1.1.1",
+    "@api3/ois": "^1.1.2",
     "@api3/promise-utils": "^0.3.0",
     "aws-sdk": "^2.1209.0",
     "body-parser": "^1.20.0",
diff --git a/packages/airnode-validator/package.json b/packages/airnode-validator/package.json
index f41ea11e..88834027 100644
--- a/packages/airnode-validator/package.json
+++ b/packages/airnode-validator/package.json
@@ -23,7 +23,7 @@
     "test:e2e:update-snapshot": "yarn test:e2e --updateSnapshot"
   },
   "dependencies": {
-    "@api3/ois": "^1.1.1",
+    "@api3/ois": "^1.1.2",
     "@api3/promise-utils": "^0.3.0",
     "dotenv": "^16.0.2",
     "ethers": "^5.7.0",
diff --git a/yarn.lock b/yarn.lock
index d36759bf..98d9642d 100644
--- a/yarn.lock
+++ b/yarn.lock
@@ -10,13 +10,13 @@
     "@jridgewell/gen-mapping" "^0.1.0"
     "@jridgewell/trace-mapping" "^0.3.9"
 
-"@api3/ois@^1.1.1":
-  version "1.1.1"
-  resolved "https://registry.yarnpkg.com/@api3/ois/-/ois-1.1.1.tgz#8760c3d4d35435d68a4726ac9d4d17e91e0ac02e"
-  integrity sha512-u7LPVmcstVdgNXnZ0JgWV23NA4SPnT8Uk5PkguFnmMr1Or1J7TO4Iq2lpTJnzJcwUTy52MndCfEMVvGvHJ5Pww==
+"@api3/ois@^1.1.2":
+  version "1.1.2"
+  resolved "https://registry.yarnpkg.com/@api3/ois/-/ois-1.1.2.tgz#104dbbe38c864868fa99a7c979230ba992651ee3"
+  integrity sha512-kKF/GtmVewrdhKd/mnAX/UCGs508dcqP/Np/FG7B8I3fX6LAH1Gg3SGXiIQZFQ0V9u8CStsf7CpZj9BVLLri+Q==
   dependencies:
     lodash "^4.17.21"
-    zod "^3.17.3"
+    zod "^3.18.0"
 
 "@api3/promise-utils@^0.3.0":
   version "0.3.0"
@@ -7472,7 +7472,6 @@ ethereumjs-abi@0.6.8, ethereumjs-abi@^0.6.8:
 
 "ethereumjs-abi@git+https://github.com/ethereumjs/ethereumjs-abi.git":
   version "0.6.8"
-  uid ee3994657fa7a427238e6ba92a84d0b529bbcde0
   resolved "git+https://github.com/ethereumjs/ethereumjs-abi.git#ee3994657fa7a427238e6ba92a84d0b529bbcde0"
   dependencies:
     bn.js "^4.11.8"
@@ -17260,3 +17259,8 @@ zod@^3.17.3:
   version "3.18.0"
   resolved "https://registry.yarnpkg.com/zod/-/zod-3.18.0.tgz#2eed58b3cafb8d9a67aa2fee69279702f584f3bc"
   integrity sha512-gwTm8RfUCe8l9rDwN5r2A17DkAa8Ez4Yl4yXqc5VqeGaXaJahzYYXbTwvhroZi0SNBqTwh/bKm2N0mpCzuw4bA==
+
+zod@^3.18.0:
+  version "3.19.0"
+  resolved "https://registry.yarnpkg.com/zod/-/zod-3.19.0.tgz#a9303fcaed0950a3dc57bf35cc11f6282d4f52f2"
+  integrity sha512-Yw0qvUsCNGBe5YacikdMt5gVYeUuaEFVDIHKMfElrGSaBhwR3CQK6vOzgfAJOjTdGIhEeoaj8GtT+NDZrepZbw==
```

**Comment by dcroote** [comment]
1. When attempting the same repo-wide OIS version bump to `1.1.2` I confirmed the build error `@api3/airnode-vali
dator: src/config/config.ts(425,10): error TS2589: Type instantiation is excessively deep and possibly infinite.`
2. However, I was able to import and use `@api3/ois@1.1.2` as a standalone package, so the issue isn't with the `
1.1.2` release in isolation
3. Bumping to `1.1.2` in `airnode-validator` and building just that worked
4. SOLVED: it is due to multiple `zod` instances in `package.json` / `yarn.lock`:

`@api3/ois@1.1.2` has `"zod": "^3.18.0"` ([here](https://github.com/api3dao/ois/blob/52fdfa7e44597979208e8ed877bf83b7363dee07/package.json#L38)), while airnode has:

```sh
packages/airnode-node/package.json:    "zod": "^3.17.3"
packages/airnode-validator/package.json:    "zod": "^3.17.3"
packages/airnode-deployer/package.json:    "zod": "^3.17.3"
```

Simultaneously bumping `zod` as in the linked PR solves the build issue.

**Comment by Siegrift** [comment]
Btw. to extend @dcroote answer...

I found it really strange that we end up with two versions of zod, because they are all compatible with each other (according to the semver). Also, I knew that [yarn is doing deduplication](https://yarnpkg.com/cli/dedupe) which should solve exactly this issue (reduce the number of packages installed)....

I researched a bit more and found out the reason for this is how yarn deduplication algorithm works... It's all nicely documented [here](https://github.com/scinos/yarn-deduplicate#duplicated-packages) and [here](https://medium.com/@bnaya/yarn-deduplicate-the-hero-we-need-f4497a362128).
